File: 0007 整数反转/main.py
import math


class Solution:
    def reverse(self, x: int) -> int:
        if x==0 or x==1:
            return x
        digits = math.floor(math.log(abs(x), 10)+1)
        rx = 0
        a = -1 if x < 0 else 1
        x = abs(x)
        for i in range(digits):
            rx *= 10
            rx += x % 10
            x = int(x/10)

        return rx*a if (-2**31 <= rx <= 2**31-1) else 0

File: 0007 整数反转/main.cpp
class Solution {
public:
    int reverse(int x) {
      int y=0;
      while (x){
        if( y > INT_MAX/10 || y < INT_MIN/10 || (y == INT_MIN/10  && x%10==-8 ||y == INT_MAX/10&& x%10 == 7) ){
            return 0;
        }
        y=y*10+x%10;
        x/=10;
      }
      return y;
    }
};